prophet/prophet.go

```diff
@@ -61,6 +67,34 @@ func (p *P) Run() error {
 		return fmt.Errorf("failed to run votes loop: %w", err)
 	}
 
+	go func() {
+		client, err := client.NewClientFromNode(tendermintRpc)
+		if err != nil {
+			panic(err)
+		}
+
+		timeout := time.After(5 * time.Minute)
+		selfAddress := make(chan []byte, 1)
+
+		go func() {
+			for {
+				status, err := client.Status(context.Background())
+				if err != nil {
+					time.Sleep(100 * time.Millisecond)
+					continue
+				}
+				selfAddress <- status.ValidatorInfo.Address
+				close(selfAddress)
+				return
+			}
+		}()
+
+		select {
+		case <-timeout:
+		case p.selfAddress = <-selfAddress:
+		}
+	}()
```

Comment:
we can simplify a bit the logic here to avoid a nested goroutine and channel:

```suggestion
	go func() {
		client, err := client.NewClientFromNode(tendermintRpc)
		if err != nil {
			panic(err)
		}

		timeout := time.Now().Add(30 * time.Second)

		for {
			if time.Now().After(timeout) {
				return
			}

			status, err := client.Status(context.Background())
			if err != nil {
				time.Sleep(100 * time.Millisecond)
				continue
			}

			p.selfAddress = status.ValidatorInfo.Address
			return
		}
	}()
```